docs/static/persistent-queues.asciidoc

```diff
@@ -101,23 +103,24 @@ queue.max_bytes: 4gb
 [[backpressure-persistent-queues]]
 ==== Handling Back Pressure
 
-Logstash has a built-in mechanism that exerts back pressure on the data flow 
-when the queue is full. This mechanism helps Logstash control the rate of data
-flow at the input stage without overwhelming downstream stages and outputs like
+When the queue is full, Logstash puts back pressure on the inputs to stall data
+flowing into Logstash. This mechanism helps Logstash control the rate of data
+flow at the input stage without overwhelming outputs and outputs like
 Elasticsearch.
 
-You can control when back pressure happens by using the `queue.max_bytes` 
-setting to configure the capacity of the queue on disk. The following example
-sets the total capacity of the queue to 8gb:
+Use `queue.max_bytes` setting to configure the total capacity of the queue on
+disk. The following example sets the total capacity of the queue to 8gb:
 
 [source, yaml]
 queue.type: persisted
 queue.max_bytes: 8gb
 
-With these settings specified, Logstash will buffer unACKed events on disk until 
-the size of the queue reaches 8gb. When the queue is full of unACKed events, and
+With these settings specified, Logstash will buffer events on disk until the
+size of the queue reaches 8gb. When the queue is full of unACKed events, and
 the size limit has been reached, Logstash will no longer accept new events. 
 
+On disk, the queue is stored as a set of pages where each page is one file. Each page can be at most `queue.page_capacity` in size. Pages are deleted (garbage collected) once all events in that page have been ACKed. If an older page has at least one event that is not yet ACKed, that entire page will remain on disk until all events in that page are successfully processed. Each page containing unprocessed events will count against the `queue.max_bytes` byte size.
```

Comment:
@jordansissel I'm curious in what scenarios this would occur. If all events in a page are successful except for one, won't that event continue to retry for a number of times until it's dropped (and therefore ack'ed)? I feel like we'd be able to solve this altogether when we have DLQs, or introducing something like compaction though that may be overkill.

It sounds like you're positioning this as an unexpected caveat/gotcha to when backpressure may be exerted. Maybe end with "Therefore, if there are unprocessed events in the queue, the 'queue.max_bytes' limit may be hit unexpectedly, exerting backpressure back to the source." Without a mechanism for queue eviction, can users work around this today aside from bumping up their queue.max_bytes?